### PR TITLE
Demo: Boot vm from ISO

### PR DESCRIPTION
**Pull request by hosh**
See issue #6

### CODE REVIEW

File: 002-vm-boot-from-iso/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Booting a VM from an ISO
+
+## Quick Start
+
+We have already prepared a VM for you to try.
```
Comment (hosh) [inline 002-vm-boot-from-iso/README.md]: Oh, this needs to be changed from VM to ISO.

File: 002-vm-boot-from-iso/README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+   1. Make sure `ipfs` daemon is running and mounted
+
+```sh
+export PATH=${HOME}/ipfs/bin:${PATH}
+ipfs daemon &
+sleep 10
+ipfs mount
+```
+
+      If you are having trouble with FUSE, please see: https://github.com/jbenet/go-ipfs/blob/master/docs/fuse.md
```
Comment (hosh) [inline 002-vm-boot-from-iso/README.md:16]: Formatting is off here for some reason.

File: 002-vm-boot-from-iso/Makefile [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+define notfound
+$(iso) not found
+Make sure the daemon is running and mounted:
+	ipfs daemon &
+	sleep 10
+	ipfs mount
+
+If you are having trouble mounting IPFS, please see https://github.com/jbenet/go-ipfs/blob/master/docs/fuse.md
+endef
+export notfound
```
Comment (jbenet) [inline 002-vm-boot-from-iso/Makefile:18]: as the makefiles get more complex, would be nice to be able to refer to these pieces of text from various places (so we can update them once). maybe this is text we can put inside files in a `000-common` dir or something. 

the downside is it's less self contained :( i like the directories are basically independent. :confused:  thoughts?

File: 002-vm-boot-from-iso/Makefile [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+	# add a hdd
+	VBoxManage createhd --filename ./disk.vdi --size 10000
+	VBoxManage storagectl $(vmname) --name "IDE Controller" --add ide
+	VBoxManage storageattach $(vmname) --storagectl "IDE Controller"  \
+	    --port 0 --device 0 --type hdd --medium ./disk.vdi
+
+	# attach the iso
+	VBoxManage storageattach $(vmname) --storagectl "IDE Controller" \
+	    --port 1 --device 0 --type dvddrive --medium "$(iso)"
```
Comment (jbenet) [inline 002-vm-boot-from-iso/Makefile:41]: maybe we should put this VBoxManage into a common shell script too? (seems like they just take a couple of parameters) making the individual makefiles simpler.

though not sure-- if we end up choosing a bunch of different params, this is likely the better approach.

**Comment by jbenet** [comment]
@hosh this LGTM. a couple comments above o/ but go for it either way. we can always change it later.

i havent tested these vms work, want me to?

(it may be useful to have travis test the demos. not sure if it can run virtualbox + VMs ..... or whether this is worth spending the effort ... but might be useful to make sure things dont break over time)

File: 002-vm-boot-from-iso/Makefile (hunk 3 above)
Comment (hosh) [inline 002-vm-boot-from-iso/Makefile:18]: I don't know. I usually don't look at how I'd organize complexity until either it is there or I have experience with it. I've rarely made Makefiles so I don't know yet what I'd want to do when there are a lot of them with snippets like that.

File: 002-vm-boot-from-iso/Makefile (hunk 4 above)
Comment (hosh) [inline 002-vm-boot-from-iso/Makefile:41]: Normally, I would be using Vagrant here.

There is only one other VM demo. That is with AstralBoot so it requires additional commands to VBoxManage to set up an isolated network.

**Comment by hosh** [comment]
@jbenet can Travis run containers?

Feel free to run them if you want, might be good to see if it works on a different computer.

Also, I have those assets pinned in my personal server right now. I would like to have you guys pin them on the Protocol Labs servers before we launch the demo -- help spread out the bandwidth costs and everything.

I may also switch everything to using `/ipns` instead of `ipfs`. It won't demo `ipfs` as an immutable structure as well though. What do you think?

File: 002-vm-boot-from-iso/Makefile (hunk 3 above)
Comment (jbenet) [inline 002-vm-boot-from-iso/Makefile:18]: sounds good!

**Comment by jbenet** [comment]
> @jbenet can Travis run containers? 

I think it can now-- or maybe it's alpha. not sure. 

> would like to have you guys pin them on the Protocol Labs servers before we launch the demo -- help spread out the bandwidth costs and everything.

Yep! Just drop the ipfs paths here: https://github.com/ipfs/infrastructure/issues/5

> I may also switch everything to using /ipns instead of ipfs. It won't demo ipfs as an immutable structure as well though. What do you think?

I think it's important to demo the immutable nature of ipfs as much as possible, and use ipns only when it makes sense to (i.e. need immutable storage). There is a lot of underrated value in knowing a link's target _will never change_.

**Comment by hosh** [comment]
This has been merged.

**Pull request closed without merging**